src/Components/Server/src/Circuits/CircuitHandler.cs

```diff
@@ -72,4 +72,29 @@ public abstract class CircuitHandler
     /// <param name="cancellationToken">The <see cref="CancellationToken"/>.</param>
     /// <returns><see cref="Task"/> that represents the asynchronous execution operation.</returns>
     public virtual Task OnCircuitClosedAsync(Circuit circuit, CancellationToken cancellationToken) => Task.CompletedTask;
+
+    /// <summary>
+    /// Invoked when an unhandled exception is thrown in the circuit.
+    /// </summary>
+    /// <param name="circuit">The <see cref="Circuit"/>.</param>
+    /// <param name="exception">The unhandled <see cref="Exception"/>.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/>.</param>
+    /// <returns><see cref="Task"/> that represents the asynchronous execution operation.</returns>
+    public virtual Task OnUnhandledExceptionAsync(Circuit circuit, Exception exception, CancellationToken cancellationToken) => Task.CompletedTask;
+
+    /// <summary>
+    /// Invoked when a JS to .NET invocation starts.
+    /// </summary>
+    /// <param name="circuit">The <see cref="Circuit"/>.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/>.</param>
+    /// <returns><see cref="Task"/> that represents the asynchronous execution operation.</returns>
+    public virtual Task OnInvokeDotNetFromJSAsync(Circuit circuit, CancellationToken cancellationToken) => Task.CompletedTask;
```

Comment:
I'm thinking it might be useful to provide additional information about what .NET method was invoked, what the arguments were, etc.